### PR TITLE
Issue/2657

### PR DESCRIPTION
**Pull request by zigouras**
**Description:**

Fix for issue #2657 Negative query tags with a minus sign on second tag does not work if it is on the same tag type than the first tag 

**Related issues and discussion:** #2657

### CODE REVIEW

File: t/leak-trace/out/regex-ampersand-report.txt [hunk 1]
```diff
@@ -0,0 +1,123 @@
+pwd:/Users/nico/projects/openfoodfacts-server/lib/ProductOpener
```
Comment (stephanegigandet) [inline t/leak-trace/out/regex-ampersand-report.txt]: This file is not part of the PR, right?

**Comment by hangy** [comment]
And please make sure to squash these changes. 🙂

**Comment by zigouras** [comment]
I am going to close this PR and redo it.  Things got messed up with trying to squash commits and remove the unneeded file.
